Avoid crash if there is a VRF terminal unit input error

Myoldmopar opened the pull request:
Addresses #4952

rraustad committed:
Protect code in GetInput when TU is not listed in TUList

rraustad committed:
Try resolving merge conflict from cherry-pick, needs review

Myoldmopar committed:
Update unit test file from addedunittest commit

rraustad committed:
Merged with develop

rraustad commented on tst/EnergyPlus/unit/HVACVariableRefrigerantFlow.unit.cc:
```diff
@@ -714,13 +703,21 @@ namespace EnergyPlus {
-
-		static std::string const RoutineName( "VRFTest_WaterCooled" );
+	TEST_F( HVACFixture, VRFTest_GetInputFailers ) {
+		// Author: R. Raustad, FSEC
```
This looks like it deleted a previous unit test. I'll add back in uisng a different branch.

rraustad commented on src/EnergyPlus/HVACVariableRefrigerantFlow.cc:
```diff
@@ -1156,6 +1156,53 @@ namespace HVACVariableRefrigerantFlow {
+		}
+
+	}
+
```
Moved up the fatal error so unit tests could call the new GetVRFInputData without hitting a fatal error.

rraustad commented on src/EnergyPlus/HVACVariableRefrigerantFlow.cc:
```diff
@@ -2289,8 +2335,12 @@ namespace HVACVariableRefrigerantFlow {
+						TerminalUnitList( VRFTU( VRFTUNum ).TUListIndex ).CoolingCoilAvailSchPtr( VRFTU( VRFTUNum ).IndexToTUInTUList ) = GetDXCoilAvailSchPtr( DXCoolingCoilType, cAlphaArgs( 12 ), errFlag );
+					} else {
+						VRFTU( VRFTUNum ).CoolingCoilPresent = false;
+					}
```
crux of issue, 2 places. Protect for 0 indexes to TerminalUnitList.

rraustad commented:
Ready to go when CI completes

Myoldmopar commented:
@rraustad This is looking good.  There are a bunch of unused variable warnings.  Can you clean them up?  If you want to add that missing unit test you mentioned above at the same time, it might be convenient.

rraustad commented:
Quick, before these show fail...

rraustad committed:
Merge remote-tracking branch 'remotes/origin/develop' into 4952-FixBranchAgain

Conflicts:
	src/EnergyPlus/HVACVariableRefrigerantFlow.cc
	tst/EnergyPlus/unit/HVACVariableRefrigerantFlow.unit.cc

rraustad committed:
1) removed unused variables
2) added in missing unit test
3) revised new VRF model to avoid potential crashes

rraustad committed:
More unused variables

rraustad commented:
Clean and Green. GetInput changes to avoid crash. Should be ready to merge after minimal review.

rraustad commented on tst/EnergyPlus/unit/HVACVariableRefrigerantFlow.unit.cc:
```diff
@@ -1051,33 +1048,17 @@ TEST_F( HVACFixture, VRFTest_SysCurve ) {
-		static std::string const RoutineName( "VRFTest_WaterCooled" );
+	TEST_F( HVACFixture, VRFTest_SysCurve_GetInputFailers ) {
+		// Author: R. Raustad, FSEC
+		
```
The unit tests keep getting stomped on. This shouldn't hold up the commit. I can come around and fix once the VRF bugs are checked in.

mjwitte commented on tst/EnergyPlus/unit/HVACVariableRefrigerantFlow.unit.cc:
```diff
@@ -1635,69 +1618,692 @@ TEST_F( HVACFixture, VRFTest_SysCurve_WaterCooled ) {
+
+	}
+
+	TEST_F( HVACFixture, VRFTest_SysCurve_WaterCooled ) {
```
@rraustad I think the unit tests are ok, just the diff is being stupid because they are so similar.  Check the full file in this branch, but I think the diff is showing the existing test being changed to the new test and then the old test being added back in here.

mjwitte commented on tst/EnergyPlus/unit/HVACVariableRefrigerantFlow.unit.cc:
```diff
@@ -1881,18 +2487,18 @@ TEST_F( HVACFixture, VRFTest_SysCurve_WaterCooled ) {
+		DataSizing::ZoneSizingRunDone = true;
+		DataSizing::ZoneEqSizing( CurZoneEqNum ).DesignSizeFromParent = false;
+		DataSizing::ZoneEqSizing( CurZoneEqNum ).SizingMethod.allocate( 25 );
+		DataSizing::ZoneEqSizing( CurZoneEqNum ).SizingMethod( DataHVACGlobals::SystemAirflowSizing ) = DataSizing::SupplyAirFlowRate;
```
@rraustad Are you intentionally adding the namespaces to these?  They aren't used that way in the other unit test in this file.  I'm in the midst of cleaning up so stray indentation changes in the main routine and will clean these up also if you agree.

rraustad replied:
I don't care either way but I thought it better highlighted which 
variable was used.

On 2015-09-09 16:14, Michael J. Witte wrote:

> In tst/EnergyPlus/unit/HVACVariableRefrigerantFlow.unit.cc [1]:
> 
> > @@ -1881,18 +2487,18 @@ TEST_F( HVACFixture,
> > VRFTest_SysCurve_WaterCooled ) {
> > DataSizing::CurZoneEqNum = 1;
> > DataEnvironment::OutBaroPress = 101325; // sea level
> > DataZoneEquipment::ZoneEquipInputsFilled = true; // denotes zone
> > equipment has been read in
> > - StdRhoAir = PsyRhoAirFnPbTdbW( DataEnvironment::OutBaroPress,
> >   20.0, 0.0 );
> > - ZoneEqSizing.allocate( 1 );
> > - ZoneSizingRunDone = true;
> > - ZoneEqSizing( CurZoneEqNum ).DesignSizeFromParent = false;
> > - ZoneEqSizing( CurZoneEqNum ).SizingMethod.allocate( 25 );
> > - ZoneEqSizing( CurZoneEqNum ).SizingMethod(
> >   DataHVACGlobals::SystemAirflowSizing ) =
> >   DataSizing::SupplyAirFlowRate;
> > - DataEnvironment::StdRhoAir = PsyRhoAirFnPbTdbW(
> >   DataEnvironment::OutBaroPress, 20.0, 0.0 );
> > - DataSizing::ZoneEqSizing.allocate( 1 );
> > - DataSizing::ZoneSizingRunDone = true;
> > - DataSizing::ZoneEqSizing( CurZoneEqNum ).DesignSizeFromParent =
> >   false;
> > - DataSizing::ZoneEqSizing( CurZoneEqNum ).SizingMethod.allocate( 25
> >   );
> > - DataSizing::ZoneEqSizing( CurZoneEqNum ).SizingMethod(
> >   DataHVACGlobals::SystemAirflowSizing ) =
> >   DataSizing::SupplyAirFlowRate;
> 
> @rraustad [2] Are you intentionally adding the namespaces to these?
> They aren't used that way in the other unit test in this file. I'm in
> the midst of cleaning up so stray indentation changes in the main
> routine and will clean these up also if you agree.
> 
> ## 
> 
> Reply to this email directly or view it on GitHub [3].
> 
>   *
> 
> ## Links:
> 
> [1] https://github.com/NREL/EnergyPlus/pull/5164#discussion_r39090211
> [2] https://github.com/rraustad
> [3] https://github.com/NREL/EnergyPlus/pull/5164/files#r39090211

mjwitte commented:
@rraustad I'm adding the VRF IDD chages to this branch.

mjwitte committed:
VRF field name cleanup, and idd tabs and trailing spaces except notes

mjwitte committed:
VRF fix error message if terminal unit list not found

mjwitte committed:
Merge remote-tracking branch 'remotes/origin/develop' into 4952-FixBranchAgain

mjwitte committed:
VRF indentation cleanup

mjwitte commented:
@rraustad Didn't see any defect files so I created a couple from a VRF example file.  One had the terminal unit list deleted from the file.  It crashed in one of the later error messages.  Fix for that checked in here.  Also updated some of the new VRF field names (initially to get rid of the two-line field name on N20). @rongpengzhang hope you agree with the changes.  Also cleaned out some tabs and lots of trailing spaces in the IDD.  Did not get rid of trailing spaces in \notes, because they might be useful when notes are concatenated.(?)   Also merged in develop.  Will let this go another round of CI tests.

mjwitte committed:
Merge remote-tracking branch 'remotes/origin/develop' into 4952-FixBranchAgain

Conflicts:
	idd/Energy+.idd.in

mjwitte commented:
Merge conflict fixed.
